src/main/java/com/flightstats/hub/app/PermissionsChecker.java

```diff
@@ -0,0 +1,19 @@
+package com.flightstats.hub.app;
+
+import com.flightstats.hub.exception.ForbiddenRequestException;
+
+public class PermissionsChecker {
+
+    public static void checkReadOnlyPermission(String failureMessage) {
+        if (HubProperties.isReadOnly()) {
```

Comment:
I don't especially like situations where throwing an exception is not...exceptional, if that makes sense. I'm not really sure what the best approach to changing that would be, so I think I might just be griping.

Reply:
I guess if I felt strongly about this in my own code I would see if I could pass a supplier (or, less preferably a runnable) into the method and this would just wrap it or return ...a response object of some kind? Not sure. Bleh, nevermind.

Reply:
Yeah, I looked at these as safety valves.  It's nice to be sure something can't write, even if code changes elsewhere in the future attempt to.  
One option is to split CUD from R all the way up to the Resources, so the whole write path just doesn't exist in a r/o node...with safety checks to ensure it can't be instantiated.